### ISSUE TITLE
Home -> ButtonsPaletteChooser: Incorrect Heading Hierarchy

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#754]
The first child control of the modal is a heading with the dialog title.  This is correct, but the heading is marked up at level 3.  For screen reader users, modal dialogs are separated from the content of the parent page, presented as standalone documents.  As such, the heading hierarchy should start at level 1. 
 
The individual button palettes are each preceded by a heading, but these are marked up at level 5.  Heading levels should not be skipped in the hierarchy, so these should be corrected to level 2.

Suggested Solution Heading Placement 
 
Instead of having the palette label included twice, once before the buttons inside a <span> and once after as a heading, just render it once as a heading before the list of buttons.  This will require modifying the MathButtonsGroup component, which is currently responsible for rendering the label as a < span >.

**Comment by jscholes** [comment]
@rupeshparab - the heading levels have been fixed, but each `<h2>` has the same ID ("paletteLabel").  This makes them unusable as accessible labels for the button lists.

In addition, the previous `aria-labelledby` attributes have been removed from the list containers, so the lists themselves don't have accessible names.  You'll need to generate and apply a unique id attribute for each heading, and then pass that down to MathButtonsGroup as a prop.

**Comment by rupeshparab** [comment]
I have made the ids unique and added the aria-labelledby attribute to the list container

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/01b7937e301676c3e612dba6d5ff84e48b3279a9.
